### PR TITLE
Chore: Algorithm research and change to .md

### PR DESCRIPTION
**Pull request by tbplong**

### CODE REVIEW

**Review by TruongMarco1305 — CHANGES_REQUESTED** [review]
Please let me know what r u doing bruh :>

**Comment by tbplong** [comment]
Research heuristic algorithm, give am overview about that, its advantages and disadvantages and change the file type from .txt to .md for better readability

File: research/algorithm1.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+3. **Iterative Refinement**: Continuously adjusting placements to improve the layout and reduce leftover gaps.
+
+### **Key Components of Heuristic Approaches in Cutting Stock Problems**
+
+1. **Product Sorting**: Prioritizes products based on size or shape. Larger products are often placed first to reduce large, irregular gaps.
+2. **Greedy Placement**: Uses a local optimization strategy by placing items in the "best" current spot, defined by criteria like minimal waste or proximity to edges.
+3. **Waste Minimization**: Evaluates possible placements and chooses the one that leaves the least amount of unused stock.
+4. **Flexibility in Decision-Making**: Heuristic methods often make trade-offs between speed and optimality, focusing on practical solutions that are near-optimal but computed quickly.
+
+### **Advantages of the Heuristic Approach**
```
Comment (TruongMarco1305) [inline research/algorithm1.md:19]: Please prolong this section for document purpose

Comment (tbplong) [reply research/algorithm1.md:19]: Done, and also refined the disadvantages section.

**Review by TruongMarco1305 — APPROVED** [review]
LGTM

**Pull request merged**